### DISCUSSION TITLE
the requirements for getting this app running under wine

### DISCUSSION THREAD

**Discussion by jfmherokiller** [textadventures/quest#1325] [Libraries and Code Samples]
at first the app will crash due to dotnet issues. 
these are fixed with `winetricks --force dotnet40 corefonts`

once dotnet is successfully installed the app will complain about TTS issues.

to fix this complex issue 
perform these steps
1. Collect the files you'll need to install.
    * In case the downloads won't easily start: The Download Center has a
      noscript tag with direct download links between the "Thank you for
      downloading" headline and the "Install Instructions" button.
    * Runtime:
      http://web.archive.org/web/20180324173437/https://www.microsoft.com/en-us/download/confirmation.aspx?id=27225
    * SDK:
      http://web.archive.org/web/20180304124818/https://www.microsoft.com/en-us/download/confirmation.aspx?id=27226
    * Languages (you'll need at least one):
      http://web.archive.org/web/20180304122215/https://www.microsoft.com/en-us/download/confirmation.aspx?id=27224
      * SR = speech recognition, TTS = text-to-speech

Install the runtime and sdk and choose a tts language.

Next copy the "Microsoft.Speech.dll" dll into the Quest 5 directory.

Finally Replace the Player.dll with my patched version
[Player.dll.tar.gz](https://github.com/textadventures/quest/files/9349128/Player.dll.tar.gz)

my dll replaces this line https://github.com/textadventures/quest/blob/28b466b83125efdaf9bcdec1013b81e207cbfe09/Player/Player.vb#L20
with this `Private m_speech As New Microsoft.Speech.Synthesis.SpeechSynthesizer`

Source used:
https://github.com/mk-pmb/msspeechsdk_v011_wine_howt

**Comment by ThePix** [comment]
Thanks for that. I have linked to it from the help page on Linux. I will leave this open, as I suspect it is an on-going issue.

**Comment by KVonGit** [comment]
This worked for me at the time, but it no longer works in Arch Linux or Ubuntu.

**Comment by jfmherokiller** [comment]
well I guess a lot has changed in 2 years tho I am now running arch. whats worse is the original guide seems to have disappeared.

**Comment by jfmherokiller** [comment]
for the sake of the original 2 links I think i found replacements here https://stackoverflow.com/questions/10634843/unable-to-correctly-install-ms-speech-platform-11-on-win-7-x64

https://www.microsoft.com/en-us/download/details.aspx?id=27225 (Runtime)
https://www.microsoft.com/en-us/download/details.aspx?id=14373 (SDK)
this seems to be the 3rd link 
https://www.microsoft.com/en-us/download/details.aspx?id=27224 (TTS and SR)

The stack overflow link seems to follow the same kind of setup as I figured out back then.

**Comment by KVonGit** [comment]
Does it work for you on Arch? (I use Arch, too. I only tested in Ubuntu because, well, you know... Arch is finicky.)

I don't think the Speech is my issue now. A few years ago (before this was posted, I mean), the error was about Speech. These instructions fixed that issue, though.

Now, it installs and Quest will open, but the window is all black when it opens. If I click around, it will display like a rectangle of each section I click (like Minesweeper, haha). I can click around the entire window, and eventually see everything (if I don't accidentally click a button while clicking).

So, the editor part will work if I do all that clicking around to see everything. I can even create a game, but the player will crash with an error about Windows Forms if I try to run a game.

**Comment by jfmherokiller** [comment]
oh yes I have seen that issue tho with other applications including one that made use of wpf. I never learned the source but I think it can be fixed if you make use of the latest version of wine or you make use of one of the gaming versions. 

I believe the visual issue is because either wine or the application is failing to redraw the window fully, and you are stuck with cached data. If I remember correctly it used to happen ALOT under x11 wine. For now I would say make sure vcruntime and the like are installed and maybe that will fix it.

**Comment by KVonGit** [comment]
>  it used to happen ALOT under x11 wine

Ah, that might be my problem. I'll look in to that.

---
Is it cool with you if I convert this Issue to a Discussion? (The discussions weren't open when you created this. It wasn't even an option until recently.)

**Comment by jfmherokiller** [comment]
feel free I am an old fart so I never knew they existed either.

**Comment by jfmherokiller** [comment]
well now that discussion is open lets try to make basic list of issues we and other issues have observed when getting this running under wine.

**Reply by KVonGit** [reply]
dotnet 4.8 will be required for Quest 5.9.

When I try to install that using `winetricks`, it says the package is broken. (I didn't try to force it.)

This might be worth mentioning, or it might be that I'm using an older version of WINE. (Which might be why I'm not having the issues I have on Arch.)

**Comment by KVonGit** [comment]
Quest 5.8 runs on Linux Mint 21.3. (I just had to "refresh" my hard drive, and I installed Mint instead of Arch (for a few days).)

![image](https://github.com/user-attachments/assets/5ae8b7f2-db46-41a5-aa48-60388a7cab3e)

---
![image](https://github.com/user-attachments/assets/e28cd6c4-4b20-4c80-8ea2-d4d6b176765d)

---
<pre><font color="#8AE234"><b>kv@halffast</b></font>:<font color="#729FCF"><b>~/Games/Quest/drive_c/Program Files/Quest219</b></font>$ wine --version
wine-6.0.3 (Ubuntu 6.0.3~repack-1)
</pre>

**Reply by jfmherokiller** [reply]
did it have the same visual glitches that were described or were those absent?

**Reply by KVonGit** [reply]
None of the visual glitches were present.

The only thing was the built-in game browser didn't work at all. It just stuck at "downloading." The editor and the player worked fine, though.

**Reply by jfmherokiller** [reply]
the browser probably needs iexplorer or some web app injection

**Reply by KVonGit** [reply]
Probably something like that, but I can easily live without it.[^1] I've probably downloaded 3 games that way over the past 8 years, anyway.

[^1]: If there was a way to search for games or to show games by author, that might be a horse of a different color.

I shall investigate this, though, as I already have Quest's source code handy.

**Comment by KVonGit** [comment]
UPDATE

I completely reinstalled Arch Linux, and I got Quest 5.8 running in WINE following steps from an old 2021 forum post:

```
WINEARCH=win32 WINEPREFIX="/home/$USER/Games/Quest" winetricks -q windowscodecs gdiplus corefonts dotnet40 vb6run speechsdk
WINEARCH=win32 WINEPREFIX="/home/$USER/Games/Quest" winecfg -v win7
WINEARCH=win32 WINEPREFIX="/home/$USER/Games/Quest" wine Downloads/quest580.exe
```

---
The editor and the player both work (although the in-app game download thing-a-ma-bob doesn't work, which is not an issue for me), and there are no visual issues.

I must have had something old from the AUR in my last Arch install or something.

---
Now, to find a way to get dotnet 4.8 installed to make this work with Quest 5.9 (which has not yet even reached its beta stage).

---
PS

I'm not downloading anything manually or using that patched DLL.

The winetricks scripts seem to be installing the Speech SDK stuff from the same URL (from archive.org), and Quest seems to at least think it is installed properly. I have not tested the text-to-speech functionality, though.

**Reply by Oneirical** [reply]
Thanks a ton for this fix! I spent many hours and tried various solutions, such as the CrossOver app, but this is the only one which worked for me.

**Reply by KVonGit** [reply]
My method is from this old forum post: https://textadventures.co.uk/forum/samples/topic/nwsz5gf0te6qdtfrwzdmga/quest-5-8-working-in-wine

I just do `winecfg -v win7` instead of calling `winecfg` to set it to Windows 7 in the GUI. (Although, I did read somewhere that my way doesn't work on all Linux distributions.)

---
Also, I *think* the reason I have to do it this way compared to the OP is because I have X11 installed and jfmherokiller is running Wayland.

**Comment by Ckosmick** [comment]
I have the latest version of quest running via wine. (5.9)
I am on Wine 10.9 and some 2025 build of winetricks.
Just replace dotnet40 with dotnet48 and you're set, like below:
`WINEPREFIX=~/.wine_quest winetricks -q windowscodecs gdiplus corefonts dotnet48 vb6run speechsdk`

**Reply by vurt834** [reply]
Tried that on mac os, doesn't work (did the same thing months ago)...I think it's due to Mac OS itself, not wine. I just ran Quest 5.8 on my 5.9 file...didn't work great. I imagine a game with simpler code would function better.
